las/credentials.py

```diff
@@ -1,68 +1,108 @@
+    :type auth_endpoint: str
 
     """
-    def __init__(self, credentials_path=None, access_key_id=None, secret_access_key=None, api_key=None):
-        if not all([access_key_id, secret_access_key, api_key]):
-            if not credentials_path:
-                credentials_path = expanduser('~/.lucidtech/credentials.cfg')
+    def __init__(self, credentials_path=None):
+        self._token = (None, None)
+        creds = list(self._read_from_file(credentials_path))
+        creds_from_environ = self._read_from_environ()
 
-            if exists(credentials_path):
-                access_key_id, secret_access_key, api_key = self._read_from_file(credentials_path)
-            else:
-                access_key_id, secret_access_key, api_key = self._read_from_environ()
+        for idx, value in enumerate(creds):
+            if creds_from_environ[idx] is not None:
+                creds[idx] = creds_from_environ[idx]
 
-        if not all([access_key_id, secret_access_key, api_key]):
+        if not all(creds):
             raise MissingCredentials
 
-        self.access_key_id = access_key_id
-        self.secret_access_key = secret_access_key
+        client_id, client_secret, api_key, auth_endpoint, api_endpoint = creds
+
+        self.client_id = client_id
+        self.client_secret = client_secret
         self.api_key = api_key
+        self.auth_endpoint = auth_endpoint
+        self.api_endpoint = api_endpoint
 
     @staticmethod
-    def _read_from_environ() -> Tuple[Optional[str], Optional[str], Optional[str]]:
-        access_key_id = os.environ.get('LAS_ACCESS_KEY_ID')
-        secret_access_key = os.environ.get('LAS_SECRET_ACCESS_KEY')
-        api_key = os.environ.get('LAS_API_KEY')
-
-        return access_key_id, secret_access_key, api_key
+    def _read_from_environ() -> List[Optional[str]]:
+        return [os.environ.get(k) for k in (
+            'LAS_CLIENT_ID',
+            'LAS_CLIENT_SECRET',
+            'LAS_API_KEY',
+            'LAS_AUTH_ENDPOINT',
+            'LAS_API_ENDPOINT'
+        )]
 
     @staticmethod
-    def _read_from_file(credentials_path: str) -> Tuple[Optional[str], Optional[str], Optional[str]]:
+    def _read_from_file(credentials_path: str) -> List[Optional[str]]:
+        if not credentials_path:
+            credentials_path = expanduser('~/.lucidtech/credentials.cfg')
+
+        if not exists(credentials_path):
+            return [None] * 5
+
         config = configparser.ConfigParser()
         config.read(credentials_path)
         section = 'default'
 
-        access_key_id = config.get(section, 'access_key_id')
-        secret_access_key = config.get(section, 'secret_access_key')
+        client_id = config.get(section, 'client_id')
+        client_secret = config.get(section, 'client_secret')
         api_key = config.get(section, 'api_key')
+        auth_endpoint = config.get(section, 'auth_endpoint')
+        api_endpoint = config.get(section, 'api_endpoint')
```

Comment:
If _read_from_environ() can return None for specific keys then I think this one should too.